appservice/src/main/java/nu/yona/server/analysis/rest/ActivityController.java

```diff
@@ -0,0 +1,212 @@
+package nu.yona.server.analysis.rest;
+
+import static nu.yona.server.rest.Constants.PASSWORD_HEADER;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+import java.time.LocalDate;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.hateoas.Resource;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import nu.yona.server.analysis.service.AnalysisEngineService;
+import nu.yona.server.analysis.service.DayActivityDTO;
+import nu.yona.server.analysis.service.WeekActivityDTO;
+import nu.yona.server.crypto.CryptoSession;
+import nu.yona.server.goals.rest.GoalController;
+import nu.yona.server.subscriptions.service.UserService;
+
+/*
+ * Controller to retrieve activity data for a user.
+ */
+@Controller
+@RequestMapping(value = "/users/{userID}/activity")
+public class ActivityController
+{
+	@Autowired
+	private AnalysisEngineService analysisEngineService;
+
+	@Autowired
+	private UserService userService;
+
+	@RequestMapping(value = "/week/", method = RequestMethod.GET)
+	@ResponseBody
+	public HttpEntity<PagedResources<WeekActivityResource>> getWeeksActivity(
+			@RequestHeader(value = PASSWORD_HEADER) Optional<String> password, @PathVariable UUID userID, Pageable pageable,
+			PagedResourcesAssembler<WeekActivityDTO> pagedResourcesAssembler)
+	{
+		return CryptoSession.execute(password, () -> userService.canAccessPrivateData(userID),
+				() -> new ResponseEntity<>(
+						pagedResourcesAssembler.toResource(analysisEngineService.getWeekActivity(userID, pageable),
```

Comment:
The analysis engine has no role here. It has done its work already and attached the result of its analysis to the user. I propose to create an ``ActivityService`` for this purpose.